### DISCUSSION TITLE
question about "default webbrowers view" used by android to show captive portal - not a library issue

### DISCUSSION THREAD

**Discussion by dncex** [Hieromon/AutoConnect#358] [Q&A]
I am making some OTA update tests (with the webserver).
There is a small issue, not related to this library, but just to android, at least with my phone (huawei p30). After connecting to the "esp32ap" with my android phone, android bring me to the captive portal, but with this browser, once I click to the "choose file" in the "update" page, nothing happen. If I open the captive portal with "chrome" or the default huawei brower, after I click "choose file" it actually works, showing me the possibility to select the .bin file. It also working with chrome from my win10 pc.
Of course I can deal with it form myself, but for other people that has to update the board, might be an issue, so I am wondering if someone has found a workaroud for android. Thanks in advance

**Comment by Hieromon** [comment]
What happens if to display a minimum of HTML of the following in your huawei p30 browser?

```html
<html>
<head>
<meta charset="utf-8" name="viewport" content="width=device-width,initial-scale=1">
</head>
<body>
<div>
<h3>Updating firmware<h3>
</div>
<label for="bin">Select file: </label><input type="file" id="bin" name="bin">
</body>
</htm>
```

The code above does intentionally not close the `<h3>` tag. There is a bug in *AutoConnectOTAPage.h* and is reproducing it.
https://github.com/Hieromon/AutoConnect/blob/4d179e7384cf158b4c86716236f1f22519ff1fbb/src/AutoConnectOTAPage.h#L15

Or, you can confirm the behavior with huawei browser using MDN documentation page.
https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/file
